### DISCUSSION TITLE
Can Linkderd be used as a Forward Proxy with support for GRPC-WEB ?

### DISCUSSION THREAD

**Discussion by joe-getcouragenow** [linkerd/linkerd2#4908] [General]
Its a double question.

1. Would like to run linkerd2 as ONLY a forward proxy. The GRPC Servers are written in golang, and we need to run them with linkerd2 on baremetal.  So, no K8 and no docker. 

2. Plus we need to support GRPC but also GRPC-Web being proxied by linkerd2, due to the need to allow Web based GUI interact with the system from the egde network.

I searched the Issues and code, but could not find much relating to support for GRPC-Web.

thanks !!

**Comment by olix0r** [comment]
Hi @joe-getcouragenow,

What do you actually want linkerd2 to _do_ for you? You might be able to make the proxy work outside the context of a service mesh, but this isn't really what it's designed for

Depending on your needs, you might consider looking at [linkerd 1.x](https://linkerd.io/1/index.html) or [Envoy](https://envoyproxy.io), which both provide general-purpose proxies.

**Comment by joe-getcouragenow** [comment]
Hey @olix0r Yep i know envoy well.

I am wanting to change to Linkerd2 for performance reasons :)
